### ISSUE TITLE
One Click Equalify Install: Simple Local Deployment

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#491]
## Problem
Equalify currently exists in lots of repos and some services depend on AWS. This prevents people from quickly deploying Equalify locally.

## Solution
A simple, one click install would empower under-resourced folks to become web accessibility testing professionals. 

I'm imagining a two-part buildout:

1. Build a proof of concept that makes Equalify quickly deployable on a signal enviornment. 
2. Use that proof of concept to train partners at organizations like [Access and Assistance for All](https://www.makethingsaccessible.com/aaa/) and the [Disability Innovation Hub](https://www.disabilityinnovation.com) on becoming accessibility professionals themselves. 
3. Raise additional funds to meet needs we found through our work with partners, so that Equalify can be deployed ever-more easily on an ever-wider range of machines.

## Budget
I'm roughly thinking we'll need about $5,000 to build the initial proof of concept, then $15,000 to $25,000 to build something that worked for the range of users we want to address.

## Next Steps
Tagging @heythisischris @ebertucc @azdak on this to get their thoughts, but open to anyone. Particularly, I'm interested in getting a better understanding of work and hours associated with the proof of concept.

**Comment by bbertucc** [comment]
On Slack @azdak said:

> one thing I'd say here is that this would _def_ be a complete fork. Back of envelope we have:
> - API on AWS and postgres
> - Dashboard on node
> - scan and workers on node with redis and mongo
> This stuff could all be rewritten and wrapped up in docker or something, but I think we'd probably not have a ton of code portability between the current and standalone versions (and honestly if it was standalone, it probably makes more sense to consolidate everything into a monolith anyways, for ease of installation/maintenance).
> So like... technically doable (everything is), but just a different and pretty separate product than what we've been building. And down the line I think it's going to be tough to easily fold improvements from the main branch into a standalone version. Curious to hear what other ppl think tho

@heythisischris said:

> I'd be curious about a "deploy on your own cloud" strategy. Just sign up for an AWS account, we'll have a simple CloudFormation script, and boom, you've got your own SaaS environment


My response:

@heythisischris - We couldn't require AWS because that would require users use a credit card and we're intentionally trying to target under resourced users in developing countries. 

@azdak I think we might need to do a light version.

@azdak @heythisischris is there really no way we can wrap it up in a docker project to deploy? I can imagine a deploy script running the various commands it takes to build a local environment. Seems weird if we are truly wed to AWS that closely. I mean, I get Cognito. But this version wouldn't need any authentication.

**Comment by heythisischris** [comment]
@bbertucc @azdak True, if we skip authentication, it becomes much simpler. And it helps that we've already wrapped the API with Fastify and the DB is PostgreSQL. React web app can be served via NGINX. Only problem I see is scan/workers with Redis/Mongo.

Agreed on splitting this into a lightweight version of Equalify, maintained separately from our current cloud-based Equalify.

Docker napkin math:
- React web app w/ NGINX (Port 1024)
- Fastify API (Port 1025)
- PostgreSQL DB (Port 1026)
- Redis (Port 1027)
- Mongo (Port 1028)

**Comment by bbertucc** [comment]
Yeah @heythisischris - this can basically work into what becomes the free, Open Source tier of Equalify.

**Comment by bbertucc** [comment]
While I wait to hear from the london college, I'm going to close this.

**Comment by bbertucc** [comment]
One concern is that this could distract from our roadmap focus. However, this issue does raise good concerns about Equalify Open Sourceness.

**Comment by ShirajG** [comment]
I'm interested in working on this. Reopening and assigning myself. Initial steps will be to analyze and out together a plan

**Comment by ShirajG** [comment]
@bbertucc can you reopen and assign me here?

**Comment by bbertucc** [comment]
Sweet! Done @ShirajG 

Let me know if you need a budget and what that budget would be. University College London was interested in sponsoring this work and I can get back in touch with them when and if you need a budget.

**Comment by ShirajG** [comment]
Will start scoping it this week as the permits. Thanks

-------- Original Message --------

**Comment by bbertucc** [comment]
Closing this again, as it seems like there's not a lot of momentum. Feel free to let me know if I should open it @ShirajG!
